src/Pages/TopicSelectPage/QuizListPage.tsx

```diff
@@ -0,0 +1,295 @@
+      case 'UNSOLVED':
+        return allQuizzes.filter((quiz) => !quiz.isSolved);
+      case 'ALL':
+      default:
+        return allQuizzes;
+    }
+  }, [quizListData?.quizzes, filterType]);
+
+  const allQuizzes = quizListData?.quizzes || [];
+
+  const handleQuizClick = (quizId: number) => {
+    navigate(`/quizSolve/${quizId}`);
+  };
+
+  if (isLoading) {
+    return (
+      <Container>
+        <Header hasPrevPage={true} title={topicName || '퀴즈 목록'} />
+        <QuizListContainer>
+          <LoadingText>퀴즈 목록을 불러오는 중...</LoadingText>
+        </QuizListContainer>
+      </Container>
+    );
+  }
+
+  if (error) {
+    return (
+      <Container>
+        <Header hasPrevPage={true} title={topicName || '퀴즈 목록'} />
+        <QuizListContainer>
+          <ErrorText>퀴즈 목록을 불러오는데 실패했습니다.</ErrorText>
+        </QuizListContainer>
+      </Container>
+    );
+  }
+
+  return (
+    <Container>
+      <Header hasPrevPage={true} title={topicName || '퀴즈 목록'} />
+      <QuizListContainer>
+        <QuizListTitle>{topicName} 퀴즈</QuizListTitle>
+        <QuizListDescription>총 {allQuizzes.length}개의 퀴즈가 있습니다.</QuizListDescription>
+
+        <FilterButtonContainer>
+          <FilterButton isActive={filterType === 'ALL'} onClick={() => setFilterType('ALL')}>
+            전체
+          </FilterButton>
+          <FilterButton
+            isActive={filterType === 'UNSOLVED'}
+            onClick={() => setFilterType('UNSOLVED')}
+          >
+            안 푼 문제
+          </FilterButton>
+          <FilterButton isActive={filterType === 'SOLVED'} onClick={() => setFilterType('SOLVED')}>
+            푼 문제
+          </FilterButton>
+        </FilterButtonContainer>
+
+        <QuizList>
+          {filteredQuizzes.map((quiz) => (
+            <QuizItem
+              key={quiz.quizId}
+              onClick={() => handleQuizClick(quiz.quizId)}
+              isSolved={quiz.isSolved}
+            >
+              <QuizItemHeader>
+                <QuizOrder>Q{quiz.questionOrder}</QuizOrder>
+                <DifficultyBadge difficulty={quiz.difficultyLevel}>
+                  {quiz.difficultyLevel === 'EASY'
+                    ? '쉬움'
+                    : quiz.difficultyLevel === 'MEDIUM'
+                      ? '보통'
+                      : '어려움'}
+                </DifficultyBadge>
+                {quiz.isSolved && <SolvedBadge>완료</SolvedBadge>}
+              </QuizItemHeader>
+              <QuizTitle>{quiz.questionTitle}</QuizTitle>
+            </QuizItem>
+          ))}
+        </QuizList>
+        {filteredQuizzes.length === 0 && allQuizzes.length > 0 && (
+          <EmptyMessage>
+            {filterType === 'SOLVED'
+              ? '푼 문제가 없습니다.'
+              : filterType === 'UNSOLVED'
+                ? '안 푼 문제가 없습니다.'
+                : '아직 퀴즈가 없습니다.'}
+          </EmptyMessage>
+        )}
+        {allQuizzes.length === 0 && <EmptyMessage>아직 퀴즈가 없습니다.</EmptyMessage>}
```

Comment:
복잡한 삼항 연산자 대신 `getEmptyMessage` 를 구현해서 아래와 같이 수정할 수 있을 것 같아요. `getEmptyMessage` 는 직접 구현해보시죠!

```suggestion
        {(filteredQuizzes.length === 0 || allQuizzes.length === 0) && (
          <EmptyMessage>{getEmptyMessage(filterType, allQuizzes.length > 0)}</EmptyMessage>
        )}
```

Reply:
### 문제 1: 복잡한 삼항 연산자 대신할 방법?

```tsx
{filteredQuizzes.length === 0 && allQuizzes.length > 0 && (
          <EmptyMessage>
            {filterType === 'SOLVED'
              ? '푼 문제가 없습니다.'
              : filterType === 'UNSOLVED'
                ? '안 푼 문제가 없습니다.'
                : '아직 퀴즈가 없습니다.'}
          </EmptyMessage>
        )}
        {allQuizzes.length === 0 && <EmptyMessage>아직 퀴즈가 없습니다.</EmptyMessage>}
```

현재 코드는, 푼문제 안푼문제, 전체퀴즈 탭에 따라 삼항 연산자로 비어있는 상황의 메세지를 분기처리 하고 있습니다.

이는 복잡하고, 가독성이 많이 떨어집니다. 삼항 연사자 자체를 최대한 쓰는걸 지양하는 것이 맞다고 생각합니다.

이에 멘토님 의견처럼 getemptyMessage라고 비어있는 상태 메세지 분기처리를 아예 함수로 분리하고자 합니다.

```tsx
const getEmptyMessage = (filterType: FilterType, hasQuizzes: boolean): string => {
  if (!hasQuizzes) {
    return '아직 퀴즈가 없습니다.';
  }

  switch (filterType) {
    case 'SOLVED':
      return '푼 문제가 없습니다.';
    case 'UNSOLVED':
      return '안 푼 문제가 없습니다.';
    case 'ALL':
    default:
      return '아직 퀴즈가 없습니다.';
  }
};
```

먼저 분기처리되는 상태값(필터타입) 과 퀴즈가 있는지 여부를 인자로 받습니다.

저 두가지 상태를 인자로 받아야, 원래코드처럼 굳이 퀴즈가 있는 상태와 없는 상태를 따로 체크하며 퀴즈가있을 때는 삼항연산자로 케이스를 분리하고, 없을때는 또 퀴즈가 없다는 메세지를 띄우는 복잡한 상황을 다 체크할 수 있습니다.

이후로는 단순하게 hasQuizzes 가 false 인 상황,

그리고 필터타입에따른 switch case 문으로 분기처리를 진행하였습니다.

```tsx
{(filteredQuizzes.length === 0 || allQuizzes.length === 0) && (
          <EmptyMessage>{getEmptyMessage(filterType, allQuizzes.length > 0)}</EmptyMessage>
        )}
```

이후에는 다음과 같이, 이제 퀴즈가 있는 상태 없는상태 따로 코드를 작성하지 않고, 필터링된퀴즈가 없거나 모든 퀴즈가 없을 때, 두번째 인자에는 모든 퀴즈의 존재 여부를 인자값으로 넣어줘어 깔끔하게 코드 작성이 가능하게 되었습니다.